### ISSUE TITLE
Error when validating if the name of the reference genome contains dots

### ISSUE THREAD

**Issue by bernatgel** [LooseLab/readfish#332]
Thanks for this great tool :)

When initializing ReadFish with a reference genome with more than one dot in its name, it fails to validate with error:

For example with genome: GCA_000001405.15_GRCh38_no_alt_plus_hs38d1_analysis_set.fna

```2024-02-08 12:16:44,481 readfish.targets Initialising Aligner
Traceback (most recent call last):
  File "/opt/anaconda3/envs/readfish/bin/readfish", line 8, in <module>
    sys.exit(main())
  File "/opt/anaconda3/envs/readfish/lib/python3.10/site-packages/readfish/_cli_base.py", line 61, in main
    raise SystemExit(args.func(parser, args, extras))
  File "/opt/anaconda3/envs/readfish/lib/python3.10/site-packages/readfish/entry_points/targets.py", line 524, in run
    worker = Analysis(
  File "/opt/anaconda3/envs/readfish/lib/python3.10/site-packages/readfish/entry_points/targets.py", line 190, in __init__
    self.mapper: AlignerABC = self.conf.mapper_settings.load_object("Aligner")
  File "/opt/anaconda3/envs/readfish/lib/python3.10/site-packages/readfish/_config.py", line 209, in load_object
    return obj_(**kwargs, **self.parameters)
  File "/opt/anaconda3/envs/readfish/lib/python3.10/site-packages/readfish/plugins/_mappy.py", line 60, in __init__
    self.validate()
  File "/opt/anaconda3/envs/readfish/lib/python3.10/site-packages/readfish/plugins/_mappy.py", line 94, in validate
    raise RuntimeError(
RuntimeError: Provided index file appears to be of an incorrect type - should be one of ['.fasta', '.fna', '.fsa', '.fa', '.fastq', '.fq', '.fasta.gz', '.fna.gz', '.fsa.gz', '.fa.gz', '.fastq.gz', '.fq.gz', '.mmi']
```

I've digged a bit and it seems that code of function `validate` in `_mappy.py` does not take into account that possibility and fail to validate the file extension. 

It seems to fail also with `XXXX.fna.mmi` for the index files, which as far as I know is the default extension for minimap2 indices.

It may be a problem on my end, because I'm surprised I found no info on these errors... but I'm reporting it anyway.

System Info:
   Linux (Ubuntu)
   readfish 2023.1.1 (installed with conda)

Thanks

Bernat

**Comment by mattloose** [comment]
As it happens this recently popped up and has been fixed in #330 

If you install readfish directly from this repository now:

pip install git+https://github.com/Looselab/readfish

you should find that your reference is handled appropriately. We will push this fix to pypi later but we are combining a couple of other fixes.

Please let us know if this does not fix the issue - for now I will close this.

**Comment by bernatgel** [comment]
Thank you very much. As for now I've patched my copy of readfish and I'll
wait for the next release.

Thanks
